### PR TITLE
Fixed issue [security] #18847: Cross-Site Scripting vulnerability via loadsecurity parameter

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue [security] #18847: Cross-Site Scripting vulnerability via loadsecurity parameter
Dev: encode using Ctml
Dev: get it only by post (not by param or cookies)

### CODE REVIEW

File: application/helpers/frontend_helper.php [hunk 1]
```diff
@@ -2082,8 +2082,10 @@ function display_first_page($thissurvey, $aSurveyInfo)
         $thissurvey['EM']['ScriptsAndHiddenInputs'] .= \CHtml::hiddenField('token', $_SESSION['survey_'.$surveyid]['token'], array('id'=>'token'));
     }
 
-    if (isset($loadsecurity)) {
-        $thissurvey['EM']['ScriptsAndHiddenInputs'] .= "\n<input type='hidden' name='loadsecurity' value='$loadsecurity' id='loadsecurity' />\n";
+    /* Get the loadsecurity parameters from captcha only by POST value and set it in current page (1st position) */
+    $loadsecurity = App()->getRequest()->getPost('loadsecurity');
+    if (!empty($loadsecurity)) {
+        $thissurvey['EM']['ScriptsAndHiddenInputs'] .= \CHtml::hiddenField('loadsecurity', $loadsecurity, array('id' => 'loadsecurity'));
```
Comment (olleharstedt) [inline application/helpers/frontend_helper.php:2088]: Hm CHtml has built-in protection for XSS issues?

Comment (olleharstedt) [reply application/helpers/frontend_helper.php:2088]: You can remove the `\` slash btw, we're already in global namespace. :)

Comment (olleharstedt) [reply application/helpers/frontend_helper.php:2088]: Oh same error everywhere. Pah. Pah!

Comment (Shnoulle) [reply application/helpers/frontend_helper.php:2088]: > Oh same error everywhere. Pah. Pah!

In 5 and 6

Comment (Shnoulle) [reply application/helpers/frontend_helper.php:2088]: > > Hm CHtml has built-in protection for XSS issues?
> 
> yes … encode all value of atributes.

https://www.yiiframework.com/doc/api/1.1/CHtml#renderAttributes-detail

**Comment by tiborpacalat** [comment]
I have tested this on my local.
Testing protocol:
1. reproduce the issue - set up survey site and trigger alert by adding `&loadsecurity=%27%3E%3Cimg%20src=x%20only=1%20onerror=alert(document.domain)%3E` at the end of survey url
2. the alert is triggered
3. switch to this branch and load the same survey with arguments above
4. the alert is not triggered

**Pull request merged**